frontend/src/contexts/AuthContext.js

```diff
@@ -1,47 +1,47 @@
 import React, { createContext, useState, useEffect } from 'react'
-import { node } from 'prop-types'
+import { useRouter } from 'next/router'
+import { node, string, object } from 'prop-types'
 
-import { logout } from '@utils/giteaInternalApi'
+import { logout, getSession } from '@utils/giteaInternalApi'
 
 export const AuthContext = createContext({
   isAuthenticated: false,
   user: null,
-  logout: async () => true,
+  logout: async () => false,
+  setUser: () => {},
+  setCsrf: () => {},
   csrf: '',
 })
 
-const AuthProvider = ({ children }) => {
-  const [auth, setAuth] = useState(false)
-  const [authenticatedUser, setAuthenticatedUser] = useState(null)
-  const [csrf, setCsrf] = useState('')
+const AuthProvider = ({ children, initialUser, initialCsrf }) => {
+  const [user, setUser] = useState(initialUser)
+  const [csrf, setCsrf] = useState(initialCsrf)
 
-  useEffect(() => {
-    setCsrf(window.session._csrf)
-  }, [authenticatedUser, auth])
-
-  const authorize = user => {
-    setAuth(true)
-    setAuthenticatedUser(user)
-  }
-
-  const deAuthorize = () => {
-    setAuth(false)
-    setAuthenticatedUser(null)
-  }
+  const { push } = useRouter()
 
   useEffect(() => {
-    const session = window?.session
-
-    if (session?.user) {
-      authorize(session.user, session._csrf)
-    } else {
-      deAuthorize()
+    if (typeof window !== 'undefined') {
```

Comment:
What is the purpose of this check? Hooks don't run on the server-side.

Reply:
Paranoia